docs/source/tutorials/describing_tasks.ipynb

```diff
@@ -1,20 +1,563 @@
 {
```

Comment:
<p>I am assuming we are still missing the last part? Should we introduce post_exec as a way to perform an arbitrary action at runtime based on the result of the task?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/radical-cybertools/radical.pilot/pull/2879/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='27'/>

Reply:
<p>Is that question prompted because you see this useful to the users or because you see a need for the tutorial to be comprehensive and complete?</p><p> </p><p>We have very little use cases for <code>post_exec</code> in RP (other than in RE!), and I have not seen it used in the wild really, so I think the added value of including it in the tutorial would be small at best.</p>

Reply:
Only for the user and I was indeed referring to the use cases we lately discussed in the context of RP. Happy to follow your lead and if you think it would not be useful not adding it. About the first question: `I am assuming we are still missing the last part? ` Is that the case?

Reply:
Yeah, I would leave the post-exec out here.

> About the first question: I am assuming we are still missing the last part? Is that the case?

I am sorry that I seem to have trouble to get on the same page - really am not trying to be difficult...  What is `the last part` we would be missing?  Do you mean some text to conclude the tutorial?  Or does the NB not cover what you think it should cover?  Or something else entirely?

Reply:
C&P from your notebook:
```
Environment Setup with named_env 

When the same environment is used for many tasks, then the collective sum of theit pre_exec activities can create a significant runtime overhead, both on the shared filesystem and also on the system load.  named_env addresses that problem: applications can prepare a task environment and then use the named_env attribute to exctivate it for the task.  This process is very lightweight on system load and runtime overhead, but has some limitations we will explain after the example below:

pilot.prepare_env(env_name='test_env', 
                  env_spec={'type' :  'venv', 
                            'setup': ['psutil']})

td = rp.TaskDescription({'executable' : '/bin/sh',
                         'arguments'  : ['-c', 'which python3; pip list | grep psutil'],
                         'named_env'  : 'test_env'
                        })
task = tmgr.submit_tasks(td)
tmgr.wait_tasks([task.uid])
print(task.stdout)

report.header('finalize')
session.close()
```
End of notebook, but the user is still waiting for "but has some limitations we will explain after the example below:".

Reply:
Ah, thank you -  that's a hanging remark indeed, will fix.